src/libraries/System.Collections/src/System/Collections/Generic/Stack.cs

```diff
@@ -31,6 +31,7 @@ public class Stack<T> : IEnumerable<T>,
         private int _version; // Used to keep enumerator in sync w/ collection. Do not rename (binary serialization)
 
         private const int DefaultCapacity = 4;
+        private const int MaxArrayLength = 0X7FEFFFFF; // This is the maximum array length value from Array.MaxArrayLength
```

Comment:
This seems out of place but there is certainly precedent elsewhere, e.g. https://github.com/dotnet/runtime/blob/1d9e50cb4735df46d3de0cee5791e97295eaf588/src/libraries/Common/src/System/Collections/Generic/EnumerableHelpers.cs#L57

Reply:
This can go away once https://github.com/dotnet/runtime/pull/43301 lands.

Reply:
Fixed by moving this temp value to nearby usage scope.